apps/events/views/speaker.py

```diff
@@ -1,11 +1,97 @@
-from rest_framework.viewsets import ModelViewSet
+from django.utils.decorators import method_decorator
+from drf_yasg.utils import swagger_auto_schema
 from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.viewsets import ModelViewSet
 
 from apps.events.models.speaker import Speaker
 from apps.events.serializers.speaker_serializer import SpeakerSerializer
+from apps.users.user_permissions import IsOrganizer
+from utils.user_utils import get_connected_user
 
 
+@method_decorator(
+    name="create",
+    decorator=swagger_auto_schema(
+        tags=["Speakers"],
+        security=[{"Bearer": []}],
+        operation_summary="Create a new speaker",
+    ),
+)
+@method_decorator(
+    name="list",
+    decorator=swagger_auto_schema(
+        tags=["Speakers"],
+        security=[],
+        operation_summary="List all speakers",
+    ),
+)
+@method_decorator(
+    name="retrieve",
+    decorator=swagger_auto_schema(
+        tags=["Speakers"],
+        security=[],
+        operation_summary="Get a speaker",
+    ),
+)
+@method_decorator(
+    name="update",
+    decorator=swagger_auto_schema(
+        tags=["Speakers"],
+        security=[{"Bearer": []}],
+        operation_summary="Update a speaker",
+    ),
+)
+@method_decorator(
+    name="partial_update",
+    decorator=swagger_auto_schema(
+        tags=["Speakers"],
+        security=[{"Bearer": []}],
+        operation_summary="Update a speaker",
+    ),
+)
+@method_decorator(
+    name="destroy",
+    decorator=swagger_auto_schema(
+        tags=["Speakers"],
+        security=[{"Bearer": []}],
+        operation_summary="Delete a speaker",
+    ),
+)
```

Comment:
```suggestion
```

Reply:
I used this format, because at the beginning I would not want to implement theses methods

Reply:
It will be better to be consistent throughout the existing code